src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -36,13 +38,56 @@
      *
      * @return Reference[]
      */
-    private function findAndSortTaggedServices($tagName, ContainerBuilder $container)
+    private function findAndSortTaggedServices($tagName, ContainerBuilder $container, string $indexAttribute = null, string $defaultIndexMethod = null)
     {
         $services = array();
 
+        if (null === $indexAttribute && null !== $defaultIndexMethod) {
+            throw new InvalidArgumentException('Default index method cannot be used without specifying a tag attribute.');
+        }
+
         foreach ($container->findTaggedServiceIds($tagName, true) as $serviceId => $attributes) {
             $priority = isset($attributes[0]['priority']) ? $attributes[0]['priority'] : 0;
-            $services[$priority][] = new Reference($serviceId);
+
+            if (null === $indexAttribute && null === $defaultIndexMethod) {
+                $services[$priority][] = new Reference($serviceId);
+
+                continue;
+            }
+
+            if (isset($attributes[0][$indexAttribute])) {
```

Comment:
here we potentially ignore a possible key from subsequent attributes, e.g. attributes[n].<index_by>, is it expected?

Reply:
That's how many passes already behave, including `AddConsoleCommandPass` and all is fine, nobody asked for something different. We shouldn't invent a use case when none has been reported :)